pkg/controller/deployment/deployment_controller.go

```diff
@@ -1237,6 +1257,18 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	if !reflect.DeepEqual(deploymentutil.GetNewReplicaSetTemplate(deployment), rs.Spec.Template) {
 		glog.Infof("Rolling back deployment %s to template spec %+v", deployment.Name, rs.Spec.Template.Spec)
 		deploymentutil.SetFromReplicaSetTemplate(deployment, rs.Spec.Template)
+		// copy RS (the old RS we'll rolling back to) annotations back to the deployment;
+		// otherwise, the deployment's current annotations (should be the same as current new RS) will be copied to the RS after the rollback.
+		//
+		// For example,
+		// A Deployment has old RS1 with annotation {change-cause:create}, and new RS2 {change-cause:edit}.
```

Comment:
How are annotations that do not directly override each other handled in this situation? e.g.
{rs1-annotation:foo}
{rs2-annotation:bar}

When we rollback to rs1 from rs2, this will add back {rs1-annotation:foo}, but does {rs2-annotation:bar} need to be explicitly removed (or is that handled by other code?)